### DISCUSSION TITLE
Auto install spylib dependencies

### DISCUSSION THREAD

**Discussion by lishanl** [SatelCreative/spylib#22] [General]
Is there a way to have the `spylib` required dependencies installed when installing `spylib`? 

Some of us have probably encountered this issue that after we install spylib, the libraries used by spylib are not installed and we need to install those libraries one by one as we discover them (through ModuleNotFound error).

Today, it took me quite a bit of time to find out that a dependency [`PyJWT==1.7.1`](https://github.com/SatelCreative/st-cim/blob/85a76c28fa8a0f25fd85124a70b67ae31d46f64a/server/requirements.txt#L27) used by `st-cim` is out of date / older than the one used in spylib [`PyJWT==2.1.0`](https://github.com/SatelCreative/satel-spylib/blob/main/pyproject.toml#L16) which caused a hard to pin down bug. 

To prevent this from happening again, is there away to auto install the dependencies used by `spylib`? Then, those requirements no longer needs to be listed in the repo that uses `spylib`?

**Comment by hillairet** [comment]
That's weird. I thought this was all handled by poetry. Something must be off. @Nathan-Nesbitt can you look into it? It should be easy to reproduce. Essentially install using `pip` locally and see what is happening.
@lishanl This could be an issue directly.

**Reply by lishanl** [reply]
@hillairet yea, I have been doing `pip install -r requirements.txt` in virtual environment. To test it more clearly, I only have `spylib==0.2.1` in the  requirements.txt and it only installs `spylib`

**Comment by estellajaysong** [comment]
I am having the same issue!

**Comment by hillairet** [comment]
@lishanl @estellajaysong Can you confirm that 0.2.2 doesn't have this problem?

**Reply by lishanl** [reply]
Sorry. missed this. I will check

**Reply by lishanl** [reply]
tested, 0.2.2 didn't fix it but 0.3 did. @hillairet  FYI

**Reply by hillairet** [reply]
Thanks for confirming that 0.3 is correct!

**Comment by qw-in** [comment]
Not sure if discussions can be converted to issues but this seems like an issue

**Reply by lishanl** [reply]
we can create issue from discussion though not directly converting it.

**Comment by hillairet** [comment]
This has been resolved. Let's not use discussions for bugs again cause we don't seem to be able to close it.
